src/Services/QuestionnaireService.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+namespace OpenEMR\Services;
+
+use OpenEMR\Common\Uuid\UuidRegistry;
+
+class QuestionnaireService extends BaseService
+{
+    public const TABLE_NAME = 'questionnaire_repository';
+
+    /**
+     * Default constructor.
+     */
+    public function __construct($base_table = null)
+    {
+        parent::__construct($base_table ?? self::TABLE_NAME);
+    }
+
+    public function fetchQuestionnaireResource($name, $q_id = null, $uuid = null)
+    {
+        $sql = "Select * From questionnaire_repository Where (name IS NOT NULL And name = ?) Or (questionnaire_id IS NOT NULL And questionnaire_id = ?)";
+        $bind =  array($name, $q_id);
+        if (!empty($uuid)) {
+            $sql = "Select * From questionnaire_repository Where uuid = ?";
+            $bind =  array($uuid);
+        }
+        $response = sqlQuery($sql, $bind) ?: [];
+        if (is_array($response) && !empty($response['id'])) {
+            $response['uuid'] = UuidRegistry::uuidToString($response['uuid']);
+        }
+        return $response;
+    }
+
+    public function getQuestionnaireIdAndVersion($name, $q_id = null, $uuid = null)
+    {
+        $sql = "Select id, uuid, version From questionnaire_repository Where (name IS NOT NULL And name = ?) Or (questionnaire_id IS NOT NULL And questionnaire_id = ?)";
+        $bind =  array($name, $q_id);
+        if (!empty($uuid)) {
+            $sql = "Select id From questionnaire_repository Where uuid = ?";
+            $bind =  array($uuid);
+        }
+        $response = sqlQuery($sql, $bind) ?: [];
+        if (is_array($response) && !empty($response['id'])) {
+            $response['uuid'] = UuidRegistry::uuidToString($response['uuid']);
+        }
+        return $response;
+    }
+
+    public function fetchQuestionnaireResponses($pid, $id = null, $name = null, $q_id = null)
+    {
+        $sql = "Select * From questionnaire_response Where patient_id = ? And (name = ? Or questionnaire_id = ?)";
```

Comment:
It looks like the PR is missing the tables for questionnaire_response.

Reply:
Not implemented yet